internal/quantile/aggregate.go

```diff
@@ -76,7 +76,7 @@ func (e *E2eProcessingLatencyAggregate) Less(i, j int) bool {
 
 func (e *E2eProcessingLatencyAggregate) Add(e2 *E2eProcessingLatencyAggregate, N int) *E2eProcessingLatencyAggregate {
 	if e == nil {
-		*e = *e2
+		e = &(*e2)
```

Comment:
This copies and then assigns the (new) address of `e2`, which is eventually returned by `Add` and then assigned over the original caller.

Gimme a bit to document that this method is side-effect free (and actually make the other branch in this method act that way instead of overwriting the receiver).